### PR TITLE
Sidebar: Split block tools into menu, settings, and appearance tabs

### LINKED ISSUE #40204: Split block tools between "settings" and "appearance"
We've discussed this as far back as 5.0, but it was not yet the right time to explore. Now that we have tools more clearly delineated, it's a good time to start looking at how we can split the block inspector between settings and appearance — appearance would generally be all the common block tools registered through supports, and setting would generally be block specific tools. The earliest mockups we had used a tabbed interface for this. There are some other considerations to discuss, including attribute types and UI slots for registering controls. We should assume by default `<InspectorControls>` are of role "settings".

Mockup:

Tabbed inspector

See also: #42257, which could benefit from this.

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Fixes: https://github.com/WordPress/gutenberg/issues/40204

## What?

- Adds new Gutenberg Experiment to put the new tabs feature behind
- Splits the block inspector primarily between settings and appearance-related tools.
- The appearance tab contains tools provided via block supports.
- The settings tab contains block-specific tools rendered into the default `InspectorControls` slot.
- Updates the sidebar icon to use two new "drawer" icons taking into account RTL.

## Why?

It's another step towards decluttering the sidebar and making it more intuitive to locate and discover block tools.

## How?

- Adds new drawer icons to the icons package
- Updates sidebar icons
- Updates the `TabPanel` component to allow icon-only buttons
- Adds Gutenberg Experiment, making the new tabs opt-in
- Updates the block inspector to render the various `InspectorControls` slots into their respective tabs when the new experiment is enabled
- The same tabs have been introduced into the block inspector for both single and multi-block selections

Note: Tabs are no longer omitted should they not have any fills in their respective slots. Instead, they'd display a message illustrating the fact there are no settings/appearance tools. This helps maintain a more consistent UI for a11y purposes.

## Testing Instructions

1. Check that the sidebar icons have been updated in both the post and site editors, and confirm they get flipped when switching between LTR and RTL.
2. In the post editor, add a navigation block, several paragraphs, a group block etc., to a post, save and select one.
3. Confirm the block inspector displays as normal.
4. Navigate to the Gutenberg Experiments admin page and enabled the Block Inspector Tabs experiment
    Screen Shot 2022-10-26 at 6 05 27 pm
5. Switch back to the editor, reload the page, and select each block.
6. Check that both tabs (settings & appearance) are in the sidebar and controls continue to function.
7. Select multiple blocks of different types and confirm no controls within either tab.
8. Select multiple blocks of the same type, e.g. two paragraph blocks. The appearance tab should still contain items.

## Screenshots or screencast 

https://user-images.githubusercontent.com/60436221/198951252-9e88ea2d-290c-4307-a1b6-09d4ef9ed9bd.mp4

### CODE REVIEW

**Comment by aaronrobertshaw** [comment]
@jasmussen and @jameskoster, any chance I can pick your brains on the best approach to improving the UX for these new sidebar tabs? 🙏 

At present, this PR is omitting any tabs that do not have anything in them. I take it that is the desired approach?

Given that, what's the desired behaviour when there are only block tools for a single tab? It looks a little odd when there is only the settings/appearance tab.

Screen Shot 2022-10-17 at 5 28 56 pm

The display of panels within the tabs probably needs some tweaking. An example of this is when there is only the advanced controls panel being shown, collapsed, under the settings tab.

Screen Shot 2022-10-17 at 5 29 04 pm

Open to any ideas you have and happy to give them a try.

**Comment by jasmussen** [comment]
Wow, fast work. This feels pretty good to me. GIF showing new subtabs, with appearance and settings, and the new inspector icon:

sidebar

Appearance should be the first tab.

> At present, this PR is omitting any tabs that do not have anything in them. I take it that is the desired approach?

I would lean towards yes, definitely, perhaps even with the addition that if we are at a point where there's only 1 tab left, remove the tabbar as well. 

> The display of panels within the tabs probably needs some tweaking. An example of this is when there is only the advanced controls panel being shown, collapsed, under the settings tab.

With settings as the 2nd tab, I think this is fine.

**Comment by jameskoster** [comment]
FWIW I don't think it would be terrible to keep the tab, but display a "no settings" message:

Screenshot 2022-10-17 at 10 17 14

Similar to how the "Block" tab itself behaves if you click it without having selected a block:

Screenshot 2022-10-17 at 10 18 36

I can imagine folks being disoriented by missing tabs ("Why can't I edit the settings for this block?"). We might consider exploring this part of the UX in a follow-up?

**Comment by mtias** [comment]
Great to see this. We should consider whether the default tab should be hinted by a block author. In the case of Paragraph we might want to default to Styles, for Navigation to List, and for Query to Settings, etc.

**Comment by mtias** [comment]
Also, if a block has no Settings we should not render any tab and just show the style groups directly.

**Comment by mtias** [comment]
Another thing that comes to mind is leveraging the Tips entry points we did a while ago for every block, as it could be shown on some of the Settings group when it's empty.

**Comment by pablohoneyhoney** [comment]
This is likely unintentional, but the sidebar/inspector icon should be inverse, no?

**Comment by aaronrobertshaw** [comment]
> This is likely unintentional, but the sidebar/inspector icon should be inverse, no?

~~@pablohoneyhoney unfortunately I can't replicate this issue. The sidebar icons are correct for me and update properly in both the block and site editors when I switch between LTR (e.g. English) and RTL (e.g. Arabic) languages.~~

EDIT: Nevermind, sorry didn't realise that was the mockup not a screenshot of this PR 🤦 

##### Block Editor
| LTR | RTL |
|---|---|
| Screen Shot 2022-10-18 at 7 02 28 pm | Screen Shot 2022-10-18 at 7 12 31 pm |



##### Site Editor
| LTR | RTL |
|---|---|
| Screen Shot 2022-10-18 at 7 18 54 pm | Screen Shot 2022-10-18 at 7 13 02 pm |

**Comment by aaronrobertshaw** [comment]
Thank you everyone for the quick feedback 🙇 

> Appearance should be the first tab

Appearance is now before the settings tab. 

While we are revisiting the order of tabs. The [mockups on the issue](https://github.com/WordPress/gutenberg/issues/40204#issuecomment-1276196445) have the Menu/List tab for the Navigation block first, should it remain like that? 

Does it make more sense to keep it located beside settings? Could blocks in future add their own tabs here?

> I would lean towards yes, definitely, perhaps even with the addition that if we are at a point where there's only 1 tab left, remove the tabbar as well.
> 
> Also, if a block has no Settings we should not render any tab and just show the style groups directly.

I've refactored things so now if there is only a single tab with contents, we render those contents directly (no TabPanel).


Having blocks define their default tab or including tips is still on the todo list.

**Comment by jameskoster** [comment]
> Could blocks in future add their own tabs here?

I would say potentially yes. One example could be transporting the concept in https://github.com/WordPress/gutenberg/issues/44582 to the template part block itself. Perhaps content-locked containers as well.

**Comment by mtias** [comment]
> Could blocks in future add their own tabs here?

Very unlikely

**Comment by aaronrobertshaw** [comment]
I've put these new tabs behind a new Gutenberg Experiment and rebased the PR to resolve the recent conflicts. The PR description has been updated accordingly.

How best to implement allowing blocks to define their own default tab might require a little iteration so I'll get up the simplest implementation for that in a separate follow-up PR and work through it there. Hopefully, that might allow us to land this sooner and test more widely.

While testing this locally with the Query block I found that it renders a "Create a new post" link into the `InspectorControls` that expects to visually look a part of the `BlockCard` at the top of the block inspector sidebar. 

| Without Tabs | With Tabs |
|---|---|
| Screen Shot 2022-10-26 at 5 59 14 pm | Screen Shot 2022-10-26 at 5 59 47 pm |
 
Would it be worth potentially adding a new slot to that `BlockCard` or immediately below it, at least? Allowing blocks, such as the Query block, to render additional info there.

**Comment by mtias** [comment]
> Would it be worth potentially adding a new slot to that BlockCard or immediately below it, at least? Allowing blocks, such as the Query block, to render additional info there.

Probably, yes. We have to be careful with how we name it to make sure it's more semantic driven and less location driven. (Example: "I want to add something relevant to the block as a whole" vs "I want to print something within the block card".)

**Comment by jameskoster** [comment]
Is there not already something available for that? I noticed that group transforms appear in the correct position:

Screenshot 2022-10-26 at 11 21 12

**Comment by aaronrobertshaw** [comment]
> Is there not already something available for that? I noticed that group transforms appear in the correct position:

Good question. Unfortunately, those block variation controls are [specifically rendered immediately following the Block Card](https://github.com/WordPress/gutenberg/blob/f5eaa4f97365d0cb385a1516b125149c839c6aeb/packages/block-editor/src/components/block-inspector/index.js#L242-L244). They don't offer a slot to render anything else into.

**Comment by aaronrobertshaw** [comment]
Just flagging on this PR that there were some [a11y concerns](https://github.com/WordPress/gutenberg/pull/45304#pullrequestreview-1156620701) raised around inconsistent behaviour between blocks when allowing block-specific default tabs. This might also give us reason to reconsider omitting tabs when they would otherwise be empty for individual blocks.

Keen to hear everyone's thoughts on this.

cc/ @alexstine

**Comment by alexstine** [comment]
The problem with multiple tabs is if they are not used for every block, users will never know where to find anything. People with vision constantly discount these small details because you get a heck of an overview of a page. It is cumbersome and annoying to use a page with a screen reader especially when every block has a different experience. My advice, leave it alone if it is not really broken. There are other projects around the editor that could best enjoy our time. I think this could be a better idea down the road but we'd need to figure out how to communicate that info better to keyboard users.

Thanks.

**Comment by aaronrobertshaw** [comment]
Appreciate the great feedback @alexstine. Thank you for raising the a11y issues here and on https://github.com/WordPress/gutenberg/pull/45304.

Would maintaining the same tabs across all blocks ([as suggested earlier in this PR](https://github.com/WordPress/gutenberg/pull/45005#issuecomment-1280555028)) alleviate the majority of these concerns? To achieve that, we might need to consolidate the navigation block's menu tab into the general settings one.

@mtias, @jasmussen, and @jameskoster, do you have any thoughts?

File: packages/block-editor/src/components/block-inspector/index.js [hunk 1]
```diff
@@ -224,11 +233,12 @@ const BlockInspector = ( { showNoBlockSelectedMessage = true } ) => {
 		<BlockInspectorSingleBlock
 			clientId={ selectedBlockClientId }
 			blockName={ blockType.name }
+			showTabs={ areBlockInspectorTabsEnabled }
 		/>
 	);
 };
 
-const BlockInspectorSingleBlock = ( { clientId, blockName } ) => {
+const BlockInspectorSingleBlock = ( { clientId, blockName, showTabs } ) => {
```
Comment (talldan) [inline packages/block-editor/src/components/block-inspector/index.js]: Very minor as this isn't a publicly exported component, but I think it'd be good to avoid adding the prop and prefer using `window?.__experimentalEnableBlockInspectorTabs` within `BlockInspectorSingleBlock`.

Just in case someone else does publicly export the component without removing this. 😅

File: packages/block-editor/src/components/inspector-controls-tabs/index.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		<InspectorControls.Slot
+			__experimentalGroup="dimensions"
+			label={ __( 'Dimensions' ) }
+		/>
+		<InspectorControls.Slot
+			__experimentalGroup="border"
+			label={ __( 'Border' ) }
+		/>
+	</>
+);
```
Comment (talldan) [inline packages/block-editor/src/components/inspector-controls-tabs/index.js]: These seem like they could all be components (`MenuTab`, `SettingsTab`, `AppearanceTab`).

File: packages/block-editor/src/components/inspector-controls-tabs/style.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.block-editor-block-inspector__tabs [role="tablist"] {
+	display: flex;
+
+	> * {
+		flex: 1;
+	}
+}
```
Comment (talldan) [inline packages/block-editor/src/components/inspector-controls-tabs/style.scss]: I think this should be added as a new 'variant' of the tab panel component.

There was a related discussion over here about making that component support different styles - https://github.com/WordPress/gutenberg/pull/44788#issuecomment-1291801144.

It could be a separate PR, but would be good to make sure it's tracked in a task list, and maybe there's an opportunity to collaborate with Jorge on it.

File: packages/block-editor/src/components/inspector-controls-tabs/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	// We have multiple tabs with contents so render complete TabPanel.
+	return (
+		<TabPanel
+			className="block-editor-block-inspector__tabs"
+			tabs={ availableTabs }
+		>
+			{ ( tab ) => {
+				if ( tab.name === TAB_MENU.name ) {
+					return getMenuTab();
+				}
```
Comment (talldan) [inline packages/block-editor/src/components/inspector-controls-tabs/index.js]: Ideally the menu tab for the navigation block isn't something that's hard coded into the block editor package.

I think it might be worth not including it for this initial PR (moving it to a separate PR so that different approaches can be discussed).

Given this is behind an experiment toggle I think it's ok for the nav menu selector to be in the normal settings tab in the first draft.

File: packages/edit-post/src/components/sidebar/settings-sidebar/index.js [hunk 5]
```diff
@@ -78,7 +78,7 @@ const SettingsSidebar = () => {
 			/* translators: button label text should, if possible, be under 16 characters. */
 			title={ __( 'Settings' ) }
 			toggleShortcut={ keyboardShortcut }
-			icon={ cog }
+			icon={ isRTL() ? drawerLeft : drawerRight }
```
Comment (talldan) [inline packages/edit-post/src/components/sidebar/settings-sidebar/index.js:81]: I'm not sure if this icon should also be part of the experiment toggle. At the moment it isn't, so merging this PR will change the icon. @jameskoster, do you think that's ok?

Personally I prefer it, I think it's much nicer than the cog. 😄

**Review by talldan — COMMENTED** [review]
This looks good to me, considering the feature is behind an experiment toggle and won't be active by default it seems pretty much ready to merge save for a couple of issues.

I think it might be good to look at Alex's feedback, and come to a decision there about how the tabs should work in this first iteration.

Also the tab specific to the navigation block (while right from a UI point of view) doesn't feel like it's implemented in the right way, and I think it could be moved to a follow-up.

IMO, the query loop issue is a minor annoyance, but isn't a blocker. If you did want to fix it though you could always add some experiment specific code to query loop block itself to make that UI appear tidier (as a temporary measure).

Before shipping as well, it'd be good to update either #40204 or a new issue with all the outstanding tasks.

**Comment by alexstine** [comment]
@aaronrobertshaw One of the best ways to have good A11Y is to not change things. Consistency is very important and often times not really included in the modern React/Vue apps. The languages are very powerful but with power comes great responsibility. Aside from bad quotes, keep things as much the same as you can, you will eliminate confusion not only for A11Y users but everyone.

File: packages/block-editor/src/components/block-inspector/index.js (hunk 1 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/block-inspector/index.js]: Good point. Done: https://github.com/WordPress/gutenberg/pull/45005/commits/433d3b20fe9ee49b204bfc34fa6c00b4ab7e780f

File: packages/block-editor/src/components/inspector-controls-tabs/index.js (hunk 2 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls-tabs/index.js]: They can and should be. Done: https://github.com/WordPress/gutenberg/pull/45005/commits/619eec79e089f2be7b73d19f710e5a0694b3e4dd

File: packages/block-editor/src/components/inspector-controls-tabs/style.scss (hunk 3 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls-tabs/style.scss]: After looking into adding a variant to the TabPanel component, I think that might be better as a follow-up to open up the chance for a more substantial refactor to improve upon a few issues we've encountered with it of late. That also ensures it doesn't delay this PR unnecessarily.

Regarding a compromise option, I've reduced the new CSS to get the tabs to fill the available space equally, down to a single line. The current approach now also removes the introduction of a new SCSS file into the mix.

I'll create an issue for exploring TabPanel improvements shortly so it doesn't slip through the cracks.

File: packages/block-editor/src/components/inspector-controls-tabs/index.js (hunk 4 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls-tabs/index.js]: Having a more focused discussion on the best approach sounds good to me. I've removed the menu tab for the navigation blocks via https://github.com/WordPress/gutenberg/pull/45005/commits/1f79403816d242fa387ea0a4ef5ac188df232586 & https://github.com/WordPress/gutenberg/pull/45005/commits/7014a44f2d50caa391ea1b9a545a25cf392a4e6b.

File: packages/edit-post/src/components/sidebar/settings-sidebar/index.js (hunk 5 above)
Comment (aaronrobertshaw) [reply packages/edit-post/src/components/sidebar/settings-sidebar/index.js:81]: I was under the impression that it was a desired tweak regardless of the sidebar tabs status. I can split the update out of this and into its own PR if necessary.

> Personally I prefer it, I think it's much nicer than the cog. 😄

Agreed. +1 on keeping it over the cog here as well.

File: packages/block-editor/src/components/inspector-controls/groups.js [hunk 6]
```diff
@@ -14,12 +14,17 @@ const InspectorControlsTypography = createSlotFill(
 	'InspectorControlsTypography'
 );
 
+// The general group is intended to render fills relevant to a block as a whole.
+// As such, they will normally be rendered outside the block inspector's tabs.
+const InspectorControlsGeneral = createSlotFill( 'InspectorControlsGeneral' );
```
Comment (aaronrobertshaw) [inline packages/block-editor/src/components/inspector-controls/groups.js]: Very open to better names for this inspector controls group. Some of the others considered included: `highlighted`, `promoted`, `important`, `block-wide`, etc.

File: packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+	return (
+		<>
+			{ ! appearanceFills.length && (
+				<span className="block-editor-block-inspector__no-block-tools">
+					{ hasSingleBlockSelection
+						? __( 'This block has no appearance tools.' )
+						: __(
+								'The selected blocks have no appearance tools.'
+						  ) }
```
Comment (aaronrobertshaw) [inline packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js]: The wording here could likely be improved. I wasn't sure about "appearance tools". Would design or block tools be any better? Any other suggestions?

**Review by aaronrobertshaw — COMMENTED** [review]
Here is a quick recap of the latest changes;

- The navigation block-specific "menu" tab has been removed; those controls now appear under the settings tab
- Each tab now has its own component
- Tabs are no longer omitted when they don't have any items; instead, they display a message as [suggested previously](https://github.com/WordPress/gutenberg/pull/45005#issuecomment-1280555028).
- A new general inspector controls group slot has been added for items that apply to a block as a whole, e.g. Query block's "create a new post" link.

<details>
<summary>Click for a demo</summary>


<video src="https://user-images.githubusercontent.com/60436221/198951252-9e88ea2d-290c-4307-a1b6-09d4ef9ed9bd.mp4" />


</details>

**Comment by aaronrobertshaw** [comment]
> @aaronrobertshaw One of the best ways to have good A11Y is to not change things. ... keep things as much the same as you can, you will eliminate confusion not only for A11Y users but everyone.

Thanks again for the continued feedback @alexstine 👍 

I've updated this PR so that tabs are not omitted when they do not have controls. Instead, such tabs will render a message illustrating that state.

Am I correct in understanding your comment about not changing things to only apply to the chopping and changing in terms of what tabs were shown? Or does it extend to not evolving the UI here?

As we are adding more and more design tools to blocks, we do have a growing need to reduce the clutter in the sidebar and help aid the discovery of related tools. I believe this was the intent behind the original issue: https://github.com/WordPress/gutenberg/issues/40204.

 I'm happy to make further tweaks to this PR to get it into a better shape for A11Y.

File: packages/edit-post/src/components/sidebar/settings-sidebar/index.js (hunk 5 above)
Comment (jameskoster) [reply packages/edit-post/src/components/sidebar/settings-sidebar/index.js:81]: Yup I think it's fine to make this change.

File: packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js (hunk 7 above)
Comment (jameskoster) [reply packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js]: It might be good to linguistically link the functionality here to global styles, IE "This block has no style options.".

File: packages/block-editor/src/components/inspector-controls/groups.js (hunk 6 above)
Comment (mtias) [reply packages/block-editor/src/components/inspector-controls/groups.js]: What's the point of this one? I don't think we should support this slot if it means what I think it means :)

File: packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js (hunk 7 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls-tabs/appearance-tab.js]: Thank you, that's much better. (https://github.com/WordPress/gutenberg/pull/45005/commits/01fbbf1eeedeb47d681c899be25650182a5538a1)

File: packages/block-editor/src/components/inspector-controls/groups.js (hunk 6 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls/groups.js]: This was the slot [touched on earlier](https://github.com/WordPress/gutenberg/pull/45005#issuecomment-1291762267) to allow items that apply to the block as a whole to be rendered above the sidebar tabs. Perhaps a better name for the inspector controls group would be `BlockInformation`?

I can see it potentially being abused by some third-party devs to get their plugin's controls a more prominent position. I've moved this slot to a separate PR ( https://github.com/WordPress/gutenberg/pull/45437 ) to discuss the best course of action further.

In the meantime, we might be able to land this PR and begin the experiment and get wider feedback. Are you happy with that plan?

**Comment by paaljoachim** [comment]
Feedback based on the initial video shared by Aaron.

Having two tab systems one within another is something I find confusing.

Current approach.
User selects a tab to show either document settings or block settings.
Within block settings there are panels to go through.

---

The new approach suggested.
User selects a tab to show either document settings or block settings.
Within Block settings there is another tab.
The result having two tab systems.

PJ explaining to a client who has barely used Gutenberg:
In your page or post. In the right sidebar top area be sure to click Block to access the block settings and styles.
When you have clicked Block tab you will need to again click the round icon just below that looks like a moon. Clicking that opens another tab. Here you can see the styles available for the block.

Screenshot 2022-11-01 at 14 48 58

One more thing. This approach brings a new tab system that sits alongside the current panels. Mixing tabs and panels. Does currently not seem like a good UX to me.

---

If I remember correctly there seems to be multiple approaches happening in regards to the sidebar. 
It would be good to discuss the various approaches to figure out a good path forward.
(Not sure what to search for to find the other issues.)

**Comment by aaronrobertshaw** [comment]
> Having two tab systems one within another is something I find confusing.

Appreciate you taking the time to provide feedback @paaljoachim 👍 

I agree there are trade-offs with the extra tabs. I still see there being some extra friction and confusion for users if we don't try to evolve the sidebar. As we are making more design tools available for blocks, that sidebar is becoming more cluttered and difficult for users to find what they need.

These new tabs are currently behind a Gutenberg experiment. This should hopefully give us a chance to seek wider feedback and allow people to actually see how it feels in action.

Are you happy with that plan?

File: packages/block-editor/src/components/inspector-controls-tabs/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+TBA...
```
Comment (talldan) [inline packages/block-editor/src/components/inspector-controls-tabs/README.md]: Lets maybe either add something basic here about the component and experiment or alternatively remove this file before merge.

**Review by talldan — APPROVED** [review]
Looks good to me as a first version! Nice work on this.

What's the best way to track the follow-up tasks for this?

**Comment by talldan** [comment]
Some other thoughts from testing:
- It doesn't feel quite right when there's only a collapsed 'Advanced' controls panel for some blocks. There is this other issue that uses the ToolsPanel approach to Advanced tools - https://github.com/WordPress/gutenberg/issues/45287. But I don't know if that will make it work any better.
- The navigation block has a `PanelColorSettings` that could be moved to the appearance tab, but would have to be careful that this doesn't negatively change the ordering when the experiment is turned off..

File: packages/block-editor/src/components/inspector-controls-tabs/README.md (hunk 8 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls-tabs/README.md]: Thanks. I've removed it until we see how the dust settles during this experiment.

**Comment by aaronrobertshaw** [comment]
> What's the best way to track the follow-up tasks for this?

Good question. Hijacking the original issue (https://github.com/WordPress/gutenberg/issues/40204) doesn't seem right, and once this PR gets merged, any list in a comment here isn't very discoverable.

I can create a separate small tracking issue if that helps.

---

FWIW the current list of follow-ups stemming from this PR includes:

- https://github.com/WordPress/gutenberg/pull/45483
- https://github.com/WordPress/gutenberg/pull/45437
- https://github.com/WordPress/gutenberg/pull/45304 (Closed due to a11y concerns)

Other possibilities are;
- Move the bespoke color controls for the Navigation block into the color panel so they show under the appearance tab
- Improving the UI around the Advanced controls when it's the only item under the settings tab (previously [decided against](https://github.com/WordPress/gutenberg/pull/45005#issuecomment-1280545225) as its impact was reduced with the settings tab appearing second)
- Create an issue for stabilizing these tabs if the experiment proves successful and desired

A little more tangential to these sidebar tabs is the possibility of refactoring the TabPanel to allow for variants. Which would allow common tab button layouts to be re-used. This is a much lower priority however, now the style changes required in this PR have been reduced to very little.

**Comment by alexstine** [comment]
@aaronrobertshaw All I need is to make sure different blocks do not adjust the sidebar dynamically. For example, if I insert a heading block and a paragraph block, the controls and tabs should be no different between the two blocks. You would not notice a problem with multiple tabs or tab differences based on the block as a visual user because you get a full overview of how things look. Screen readers rely on pure context. If every block had a possibly different set of tabs for it's own functionality, we are doing it wrong.

Thanks.

**Comment by talldan** [comment]
I've rebased this and I'm going to merge it.

Some things to note: 
- This PR is currently not affected by the accessibility discussion above, that conversation is related to https://github.com/WordPress/gutenberg/pull/45483.
- This does introduce a new way to specify an icon as part of a `TabPanel` tab. That seems ok to me. I've spent some time looking at whether `TabPanel` can be refactored and I don't think this change will cause any issues (including if the underlying implementation is refactored to ariakit).
- This is currently implemented behind an experiment toggle, so it isn't the default experience for plugin users, but I would encourage testing and feedback!

**Comment by ntsekouras** [comment]
Thanks all the work here!

I haven't checked the code here to suggest something, but just noting that Query Loop block uses a filter to add the `new post link` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query/hooks.js#L34) and it's a bit off right now.

https://user-images.githubusercontent.com/16275880/202733749-e384fc58-1f0c-4d48-a977-0cff27861d86.mov

**Comment by mtias** [comment]
@ntsekouras indeed, that is being addressed in #45437

**Comment by ntsekouras** [comment]
>@ntsekouras indeed, that is being addressed in https://github.com/WordPress/gutenberg/pull/45437

Thanks! I missed that.

**Comment by ndiego** [comment]
Just wanted to add a little follow-up that this new functionality is working great in my testing with third-party extensions. Separating the content really simplifies the sidebar panel, great work everyone! 

block-panels

For extenders, is there a way to hook into either the Styles or Settings? Looks like any extensions are put in Settings, but maybe I am missing something.

**Comment by aaronrobertshaw** [comment]
> For extenders, is there a way to hook into either the Styles or Settings?

My apologies for the slow reply @ndiego.

For now, any controls rendered into the block support panels dedicated to styling are rendered into the Styles tab i.e. border, colors, typography etc. Anything rendered to the default or advanced inspector controls slots will appear under the Settings tab.

For reference you can see exactly what is rendered for each tab via the links below:
- [Styles tab](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inspector-controls-tabs/styles-tab.js)
- [Settings tab](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inspector-controls-tabs/settings-tab.js)

As we gather feedback from this Gutenberg experiment, we'll aim to delineate what should be rendered within each tab and document the feature and our recommendations accordingly.

**Comment by ndiego** [comment]
> My apologies for the slow reply @ndiego.

No worries at all. Getting some really positive feedback on this experiment [on Twitter](https://twitter.com/nickmdiego/status/1602654262386909184). As I have played with it more, I like how only native block supports render on the styles panel. That said, I can see how people would want to add additional "style" controls to this panel, and custom blocks add another dimension. Definitely things to be worked out, but it feels like we are headed in the correct direction with this.
